crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -714,29 +718,48 @@ where
 }
 
 // TODO: categorize all possible SuiErrors
-fn convert_to_quorum_driver_error_if_nonretryable(
-    err: SuiError,
-    tx_digest: &TransactionDigest,
-    action: &'static str,
-) -> Option<QuorumDriverError> {
-    match &err {
+fn convert_to_quorum_driver_error_if_nonretryable(err: &SuiError) -> Option<QuorumDriverError> {
+    match err {
         // TODO: rewrite the equivocation detection code to make it more deterministic
         SuiError::QuorumFailedToProcessTransactionWithConflictingTransactions {
             conflicting_txes,
             retried_tx_digest,
             retried_tx_success,
+        } => Some(QuorumDriverError::ObjectsDoubleUsed {
+            conflicting_txes: conflicting_txes.clone(),
+            retried_tx: *retried_tx_digest,
+            retried_tx_success: *retried_tx_success,
+        }),
+        SuiError::QuorumFailedToProcessTransaction {
+            good_stake: _,
+            errors,
+            conflicting_tx_digests,
         } => {
-            debug!(?tx_digest, "Got unretryable error when {action}: {err}");
-            Some(QuorumDriverError::ObjectsDoubleUsed {
-                conflicting_txes: conflicting_txes.clone(),
-                retried_tx: *retried_tx_digest,
-                retried_tx_success: *retried_tx_success,
-            })
-        }
-        _ => {
-            debug!(?tx_digest, "Got retryable error when {action}: {err}");
-            None
+            if !conflicting_tx_digests.is_empty() {
+                return None;
+            }
+
+            // All the errors are the same, pick a sentinel error
+            let error = match &errors[..] {
+                [error, errors @ ..] if errors.iter().all(|e| e == error) => error,
+                _ => return None,
+            };
+
+            match error {
+                SuiError::TransactionInputObjectsErrors { errors } => match &errors[..] {
```

Comment:
I convinced myself that all `TransactionInputObjectsErrors` should be considered non-retryable. 
The gist here is that we only consider a tx non-retryable we have non-retryable response from >= f+1 validators.

I think a simpler hotfix is, go over the list of errors, if >= f+1 errors are `TransactionInputObjectsErrors` then we say this tx is terminal.

Reply:
relevant PRs: https://github.com/MystenLabs/sui/pull/7369 and https://github.com/MystenLabs/sui/pull/7343